src/sql/src/plan/query.rs

```diff
@@ -2094,7 +2094,10 @@ fn plan_scalar_table_funcs(
     let rows_from_qcx = qcx.derived_context(from_scope.clone(), qcx.relation_type(relation_expr));
 
     for (table_func, id) in table_funcs.iter() {
-        table_func_names.insert(id.clone(), table_func.name.0.last().unwrap().clone());
+        table_func_names.insert(
```

Comment:
This appears to be only tested in unit tests, might consider if it's testable in end-to-end test too.

Reply:
The change this PR introduces is how the table's identified; the original change for this work included extensive tests https://github.com/MaterializeInc/materialize/pull/9988